hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/JDBCSource.java

```diff
@@ -0,0 +1,235 @@
+    DataFrameReader dataFrameReader = null;
+    FSDataInputStream passwordFileStream = null;
+    try {
+      dataFrameReader = session.read().format("jdbc");
+      dataFrameReader = dataFrameReader.option(Config.URL_PROP, properties.getString(Config.URL));
+      dataFrameReader = dataFrameReader.option(Config.USER_PROP, properties.getString(Config.USER));
+      dataFrameReader = dataFrameReader.option(Config.DRIVER_PROP, properties.getString(Config.DRIVER_CLASS));
+      dataFrameReader = dataFrameReader
+          .option(Config.RDBMS_TABLE_PROP, properties.getString(Config.RDBMS_TABLE_NAME));
+
+      if (properties.containsKey(Config.PASSWORD) && !StringUtils
+          .isNullOrEmpty(properties.getString(Config.PASSWORD))) {
+        LOG.info("Reading JDBC password from properties file....");
+        dataFrameReader = dataFrameReader.option(Config.PASSWORD_PROP, properties.getString(Config.PASSWORD));
+      } else if (properties.containsKey(Config.PASSWORD_FILE) && !StringUtils
+          .isNullOrEmpty(properties.getString(Config.PASSWORD_FILE))) {
+        LOG.info(
+            String.format("Reading JDBC password from password file %s", properties.getString(Config.PASSWORD_FILE)));
+        FileSystem fileSystem = FileSystem.get(new Configuration());
+        passwordFileStream = fileSystem.open(new Path(properties.getString(Config.PASSWORD_FILE)));
+        byte[] bytes = new byte[passwordFileStream.available()];
+        passwordFileStream.read(bytes);
+        dataFrameReader = dataFrameReader.option(Config.PASSWORD_PROP, new String(bytes));
+      } else {
+        throw new IllegalArgumentException(String.format("JDBCSource needs either a %s or %s to connect to RDBMS "
+            + "datasource", Config.PASSWORD_FILE, Config.PASSWORD));
+      }
+
+      addExtraJdbcOptions(properties, dataFrameReader);
+
+      if (properties.containsKey(Config.IS_INCREMENTAL) && StringUtils
+          .isNullOrEmpty(properties.getString(Config.IS_INCREMENTAL))) {
+        DataSourceUtils.checkRequiredProperties(properties, Arrays.asList(Config.INCREMENTAL_COLUMN));
+      }
+      return dataFrameReader;
+    } catch (Exception e) {
+      throw new HoodieException(e);
+    } finally {
+      IOUtils.closeStream(passwordFileStream);
+    }
+  }
+
+  private static void addExtraJdbcOptions(TypedProperties properties, DataFrameReader dataFrameReader) {
+    Set<Object> objects = properties.keySet();
+    for (Object property : objects) {
+      String prop = (String) property;
+      if (prop.startsWith(Config.EXTRA_OPTIONS)) {
+        String[] split = prop.split("\\.");
+        String key = split[split.length - 1];
+        String value = properties.getString(prop);
+        LOG.info(String.format("Adding %s -> %s to jdbc options", key, value));
+        dataFrameReader.option(key, value);
+      }
+    }
+  }
+
+  @Override
+  protected Pair<Option<Dataset<Row>>, String> fetchNextBatch(Option<String> lastCkptStr, long sourceLimit) {
```

Comment:
how about use `sourceLimit` to limit num records read from RDBMS. Thats `select xxx from xxx limit sourceLimit` ?

Reply:
@leesf problem with that would be not all databases support the limit clause. Some support limit, some say top and the lingo is different. To instead just pull everything

Reply:
We could limit the df if needed but that's just added complexity on the user. Because between batches we would then have to manage what we have already sent earlier and what we should send now and in either case spark is always reading everything from RDBMS but just limiting to df level so i don't think we should be doing anything here. @vinothchandar please advise further if am wrong

Reply:
@leesf and @vinothchandar Im sure we cannot limit through sql query as I see that Mysql and Postgres do `SELET * FROM XXX LIMIT 1` where as orcale uses `SELET * FROM XXX FETCH NEXT 1 ROWS` and derby uses `SELET * FROM XXX FETCH FIRST 10 ROWS ONLY`

Reply:
Good point, in case different SQL syntax in different RDBMS, I think it is ok to ignore sourceLimit.

Reply:
Limiting might be helpful to break down the load into smaller chunks. DBMSes don't usually like large scans... So having some ability to limit would actually be good.. 

@taherk77 how about having the ability to add a `LIMIT` clause depending on the jdbc endpoint.. it should tell you if its MySQL or Postgres (two are very popular anyway, so having this working even for those 2 initially would be awesome)

Reply:
So on the driver class that the user gives straight away if it contains MySql or postgres keyword we should start applying limit?

Reply:
> Limiting might be helpful to break down the load into smaller chunks. DBMSes don't usually like large scans... So having some ability to limit would actually be good..
> 
> @taherk77 how about having the ability to add a `LIMIT` clause depending on the jdbc endpoint.. it should tell you if its MySQL or Postgres (two are very popular anyway, so having this working even for those 2 initially would be awesome)

Hi @vinothchandar  so do you mean if the user sets the limit to 10.  For postgres and MYSQL we should do select * from table limit 10?

I dont think that would work here with the type of semantics we have here. In continuous mode and full scan JDBC scans the whole table every interval. 

In incremental we first do a full scan and write checkpoints we then assume the column given for incremental is either a long, int or timestamp. If the query for incremental fails then we fall back to full scans. How would limit work here?

It would always keep getting the same records.

Further talking about interval of jobs. This has not yet implemented as I do not have clarity. I want to know how we should do it. This would require further brain storming of how to keep schedule jobs.